### ISSUE TITLE
New codegen and TypedDocumentNode

### ISSUE THREAD

**Issue by Urigo** [aaronpowell/graphql-code-generator-sample#9]
Awesome example!

Another cool thing you might want to add is to update Codegen to the latest and use the new TypedDocumentNode plugin!

Here are some related resources:

1. https://the-guild.dev/blog/typed-document-node
2. https://the-guild.dev/blog/graphql-codegen-and-apollo-client-3
3. https://graphql-code-generator.com/docs/plugins/typed-document-node

**Comment by aaronpowell** [comment]
I'll have to check it out. I'm working on a new version of this for an upcoming workshop

**Comment by Urigo** [comment]
I would love to help, just let me know if you have any questions!

**Comment by aaronpowell** [comment]
@Urigo having a look at `TypedDocumentNode`, is the primary benefit of it that you can generate more portable definitions? So instead of generating a bunch of React hooks, you create `TypedDocumentNode`s which can be used by React _or_ Angular _or_ Vue _or_ etc..

**Comment by aaronpowell** [comment]
btw, the workshop is here: https://github.com/aaronpowell/graphql-typescript-workshop

**Comment by Urigo** [comment]
Yeah, exactly! The generated TypedDocumentNode is completely agnostic to the client/framework/environment you are using. 
So there are 3 major benefits here:

1 - The `DocumentNode` has the types burned in - that means you don't have to manually specify generic types everywhere (less mistakes, more strict, easier to use) 
2 - Major libraries (like Apollo-Client and Urql) are adopting this workflow, and you can use it today. (And you can use the provided patches in case you are using other clients)
3 - The fact that we don't generate any function or variable has positive effect on bundlesize. 
4 - The usage is more similar to the original docs on the clients websites

**Comment by Urigo** [comment]
if you ever want to meet and go through it together I would love to, but from everything I see it's a great workshop!

**Comment by aaronpowell** [comment]
That's what I thought after doing some comparisons between the two outputs, there's very little difference in using `TypedDocumentNode` and the hooks, with more positive benefits.

If you're still contracting into MS, we can sync on Teams (although I think the time overlap is pretty poor between us).

**Comment by Urigo** [comment]
yes we are, just reach out!
